app/src/main/java/edu/wpi/first/shuffleboard/app/Shuffleboard.java

```diff
@@ -40,12 +42,21 @@ public void init() throws Exception {
 
   @Override
   public void start(Stage primaryStage) throws IOException {
-    primaryStage.setScene(new Scene(mainPane));
-    primaryStage.setMinWidth(640);
-    primaryStage.setMinHeight(480);
-    primaryStage.setWidth(Screen.getPrimary().getVisualBounds().getWidth());
-    primaryStage.setHeight(Screen.getPrimary().getVisualBounds().getHeight());
-    primaryStage.show();
+    try {
+      JUnique.acquireLock(getClass().getCanonicalName(), message -> {
+        primaryStage.toFront();
+        return null;
+      });
+
+      primaryStage.setScene(new Scene(mainPane));
+      primaryStage.setMinWidth(640);
+      primaryStage.setMinHeight(480);
+      primaryStage.setWidth(Screen.getPrimary().getVisualBounds().getWidth());
+      primaryStage.setHeight(Screen.getPrimary().getVisualBounds().getHeight());
+      primaryStage.show();
+    } catch (AlreadyLockedException e) {
+      JUnique.sendMessage("alreadyRunning", "true");
```

Comment:
sendMessage takes appId, which we're passing earlier as `getClass().getCanonicalName()`